linkedlist/src/main/java/ua/hillel/LinkedList.java

```diff
@@ -1,69 +1,185 @@
+    }
 
     @Override
     public void add(T element) {
-
+        Node<T> newNode = new Node<>(element);
+        if (head == null) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(int index, T element) {
-
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == size) {
+            add(element);
+            return;
+        }
+        Node<T> newNode = new Node<>(element);
+        if (index == 0) {
+            newNode.next = head;
+            head.prev = newNode;
+            head = newNode;
+        } else {
+            Node<T> node = getNodeAtIndex(index);
+            newNode.next = node;
+            newNode.prev = node.prev;
+            node.prev.next = newNode;
+            node.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> node = getNodeAtIndex(index);
+        return node.data;
     }
 
     @Override
     public T get(T element) {
+        Node<T> node = head;
+        while (node != null) {
+            if (node.data.equals(element)) {
+                return node.data;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public T getFirst() {
-        return null;
+        if (head == null) {
+            throw new NoSuchElementException();
+        }
+        return head.data;
     }
 
     @Override
     public T getLast() {
-        return null;
+        if (tail == null) {
+            throw new NoSuchElementException();
+        }
+        return tail.data;
     }
 
     @Override
     public void set(int index, T element) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> node = getNodeAtIndex(index);
+        node.data = element;
+    }
+
+    private Node<T> getNodeAtIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> node = head;
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
+        return node;
+    }
 
+    private Node<T> getNode(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> node = head;
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
+        return node;
     }
 
     @Override
     public boolean remove(int index) {
-        return false;
+        if (index < 0 || index >= size) {
+            return false;
+        }
+        Node<T> nodeToRemove = getNode(index);
```

Comment:
The same here, are you removing the correct node? Looks like you do it to the previous node. Could you please check it as well?